### ISSUE TITLE
Connecting to MemGPT

### ISSUE THREAD

**Issue by david-strejc** [enricoros/big-AGI#306]
There is project which will in future probably add "memory" to LLMs models.

They don't have API yet but it will be there soon.

Maybe implement this connection?

https://github.com/cpacker/MemGPT

**Comment by enricoros** [comment]
Thanks for the rqeust; it's hard to understand the effort until we see the API, but it's a good idea. Depending on how good the API is, we can choose whether to adopt it, or whether to mimic the same functionality.
Do you know who's gonna be in charge of the API? Historically those projects don't do as good as a job with APIs as they do with their core code.

**Comment by david-strejc** [comment]
> Thanks for the rqeust; it's hard to understand the effort until we see the API, but it's a good idea. Depending on how good the API is, we can choose whether to adopt it, or whether to mimic the same functionality. Do you know who's gonna be in charge of the API? Historically those projects don't do as good as a job with APIs as they do with their core code.

I only know it is quite a new project - probably two months old from people from Berkley - but they promised API in version 1.0. It works through commandline already - but there is probably lot of work to be done. I am watching this project because I need to connect our own data with LLMs and this is probably the only "right way" for now.

**Comment by enricoros** [comment]
Yes, it's very interesting indeed. I wonder if it's "the" way, or just a stepping stone, like embeddings, rag, and large context windows.

**Comment by enricoros** [comment]
Closing as obsolete.

Closing this on the specific integration. MemGPT is Letta now, a full agent platform rather than a memory API to bolt onto a chat app, and proxying someone else's agent runtime is not a direction we want. An external memory server is a natural MCP client case, so I have carried the ask to #892, where the memory requests are collecting - this was the most upvoted of them and that signal moves with it. Thanks for flagging it early.

@david-strejc this close comes from an automated backlog pass - if you disagree, feel free to reopen.
